src/linux/init/config.cpp

```diff
@@ -1034,23 +1034,17 @@ try
 
     if (Config.BootCommand.has_value())
     {
-        const int ChildPid =
-            UtilCreateChildProcess("BootCommand", [Command = Config.BootCommand.value(), SavedSignals = g_SavedSignalActions]() {
-                //
-                // Restore default signal dispositions for the child process.
-                //
-
-                THROW_LAST_ERROR_IF(UtilSetSignalHandlers(SavedSignals, false) < 0);
-                THROW_LAST_ERROR_IF(UtilRestoreBlockedSignals() < 0);
+        UtilCreateChildProcess("BootCommand", [Command = Config.BootCommand.value(), SavedSignals = g_SavedSignalActions]() {
+            //
+            // Restore default signal dispositions for the child process.
+            //
 
-                execl("/bin/sh", "sh", "-c", Command.c_str(), nullptr);
-                LOG_ERROR("execl() failed, {}", errno);
-            });
+            THROW_LAST_ERROR_IF(UtilSetSignalHandlers(SavedSignals, false) < 0);
+            THROW_LAST_ERROR_IF(UtilRestoreBlockedSignals() < 0);
 
-        if (ChildPid < 0)
-        {
-            LOG_ERROR("fork failed {}", errno);
-        }
+            execl("/bin/sh", "sh", "-c", Command.c_str(), nullptr);
+            LOG_ERROR("execl() failed, {}", errno);
```

Comment:
Probably for a future change, but now that we have warnings support in init, we could just an stderr warning when the boot command fails